pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java

```diff
@@ -71,6 +71,9 @@ public void shutDown() {
   public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOptions sqlNodeAndOptions,
       @Nullable RequesterIdentity requesterIdentity, RequestContext requestContext, @Nullable HttpHeaders httpHeaders)
       throws Exception {
+    if (requestContext.getRequestArrivalTimeMillis() == 0) {
```

Comment:
Might be more robust if we check for `<= 0` because the return value is per implementation based